de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/MetadataTemplateViewDialog.java

```diff
@@ -49,45 +51,54 @@
  * Created by sriram on 5/9/16.
  */
 public class MetadataTemplateViewDialog extends IPlantDialog {
-   final MetadataView.Appearance appearance = GWT.create(MetadataView.Appearance.class);
+    final MetadataView.Appearance appearance = GWT.create(MetadataView.Appearance.class);
     private final MetadataView.Presenter presenter;
     private VerticalLayoutContainer widget;
     private final DateTimeFormat timestampFormat;
     private boolean writable;
-    private final FastMap<Avu> templateAttrAvuMap;
-    private final FastMap<Field<?>> templateAttrFieldMap;
-    private List<MetadataTemplateAttribute> attributes;
-    private List<Avu> templateMd;
+    FastMap<Avu> templateTagAvuMap;
+    FastMap<MetadataTemplateAttribute> templateTagAtrrMap;
+    FastMap<Field<?>> templateTagFieldMap;
+    List<MetadataTemplateAttribute> attributes;
+    final FastMap<VerticalLayoutContainer> templateAttrVLCMap;
+    List<Avu> templateMd;
     private boolean valid;
 
-    public MetadataTemplateViewDialog(MetadataView.Presenter presenter,List<Avu> templateMd, boolean writable,
+    public MetadataTemplateViewDialog(MetadataView.Presenter presenter,
+                                      List<Avu> templateMd,
+                                      boolean writable,
                                       List<MetadataTemplateAttribute> attributes) {
-        templateAttrAvuMap = new FastMap<>();
-        templateAttrFieldMap = new FastMap<>();
+        templateTagAvuMap = new FastMap<>();
+        templateTagFieldMap = new FastMap<>();
+        templateTagAtrrMap = new FastMap<>();
+        templateAttrVLCMap = new FastMap<>();
         timestampFormat = DateTimeFormat.getFormat(DateTimeFormat.PredefinedFormat.DATE_TIME_SHORT);
         this.presenter = presenter;
         this.writable = writable;
         this.attributes = attributes;
         this.templateMd = templateMd;
         valid = true;
-        
+
         widget = new VerticalLayoutContainer();
         widget.setScrollMode(ScrollSupport.ScrollMode.AUTOY);
+        widget.getElement().applyStyles(appearance.backgroudStyle());
         buildAvuMap();
-        loadTemplateAttributes();
+        initTemplate();
         add(widget);
     }
 
     public ArrayList<Avu> getMetadataFromTemplate() {
-        ArrayList<Avu> avus =new ArrayList<>();
-        for (String attr : templateAttrFieldMap.keySet()) {
-            Avu avu = templateAttrAvuMap.get(attr);
+        ArrayList<Avu> avus = new ArrayList<>();
```

Comment:
I think ideally this logic should be in the presenter and tested?  That would eliminate the need to call `MetadataPresenterImpl.newMetadata()` and it would "dumb down" the view.